### PR TITLE
Mike icon migration

### LINKED ISSUE #123: Mike icons migration
Related to migrating the mike icons into the react-components lab and the considerations in this regard such as using the icons from a separate package and so on.

CC @rihr

### PR DESCRIPTION
**Pull request by rosudhi**
## This PR
Closes #123 

- [x] Add initial setup for icons
- [x] Add first icon
- [ ] Add other icons
- [ ] Improvements ? 

### Fulfilled the scope of this repo?
To migrate all mike's icon and perhaps to have new icons ?

### Completness

- [ ] Migrate all mike's icon

### CODE REVIEW

**Comment by rosudhi** [comment]
This icon is already have a storybook, but still needs some styling on the icons

**Review by bertearazvan — COMMENTED** [review]
I will let Rihards say more about the icons and the svg part of it as I don't have a good overview of that yet. However, I would bring these improvements:
- [ ] The icons should not have their own storybook, but mainly use the labs storybook deployment for showcasing them - or not at all if that proves to be difficult.
- [ ] This package should have it's own pipeline to deploy the library - I can pick up on that if you'd like

**Review by rihr — COMMENTED** [review]
I think there is too much work in adding an icon. (add svg, add to index.ts, add icon react component, add to storybook?) - it could even discourage sharing

Would be nicer if I could just drop in the svg file and push it, then the CI would generate and publish react icon components from that.
This should be easy to accomplish if you were to use the SVGR CLI or Node API to generate all icons found in the svg dir with the template you already made. Afterwards, the icons can be transpiled into a dist dir with babel (or maybe something else) - this would also mean no more need to edit index files.

I think a better name for the package could simply be `icons`, so the import package would be `@dhi/icons` Any thoughts @bertearazvan ?

Nice work otherwise
If you'd like I can also continue working on this to add in the suggestions

File: packages/react-components-icons/package.json [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  "typings": "dist/index.d.ts",
+  "files": [
+    "dist",
+    "src"
+  ],
+  "engines": {
+    "node": ">=10"
+  },
+  "scripts": {
+    "dev": "yarn sb",
```
[the remark this answers is not in the document]
Comment (rosudhi) [reply packages/react-components-icons/package.json:21]: yes, that's should be yarn storybook, but we don't really need that actually.

**Comment by bertearazvan** [comment]
> I think there is too much work in adding an icon. (add svg, add to index.ts, add icon react component, add to storybook?) - it could even discourage sharing

As said, I wouldn't make a storybook only for this package, but would use instead the lab storybook. If this is too difficult, we can have no storybook for this package.

> I think a better name for the package could simply be `icons`, so the import package would be `@dhi/icons` Any thoughts @bertearazvan ?

I like that

**Comment by rosudhi** [comment]
> I will let Rihards say more about the icons and the svg part of it as I don't have a good overview of that yet. However, I would bring these improvements:
> 
> * [ ]  The icons should not have their own storybook, but mainly use the labs storybook deployment for showcasing them - or not at all if that proves to be difficult.
> * [ ]  This package should have it's own pipeline to deploy the library - I can pick up on that if you'd like

yes, it would be better if every icons can be added / deployed automatically into lab component, if we want to remove storybook on it. i was thinking that everyone whose want to use the icons, can be easily look or event searching on our icons library.


> I think there is too much work in adding an icon. (add svg, add to index.ts, add icon react component, add to storybook?) - it could even discourage sharing
> 
> Would be nicer if I could just drop in the svg file and push it, then the CI would generate and publish react icon components from that.
> This should be easy to accomplish if you were to use the SVGR CLI or Node API to generate all icons found in the svg dir with the template you already made. Afterwards, the icons can be transpiled into a dist dir with babel (or maybe something else) - this would also mean no more need to edit index files.
> 
> I think a better name for the package could simply be `icons`, so the import package would be `@dhi/icons` Any thoughts @bertearazvan ?
> 
> Nice work otherwise
> If you'd like I can also continue working on this to add in the suggestions

drag-drop svg file could be nice way to encourage other to sharing their icons, but needs some rules to prevent any inappropriate to be uploaded. besides that, we already have quite a lot on Mike's icon, then the designer teams can add more later.

Conclusions:
- you guys, can freely modify the codes to achieve our goal, that's why i marked this PR as a draft. :)
- hopefully, this icons can be easy to use and as flexible as it can be. for example, we can resize or recolor the icon just by passing the props. Also anyone can easily find their wanted icon(s).
- ...

**Comment by rihr** [comment]
I don't know how the storybook could work with icons being generated as individual react components directly to the `dist` dir.

You can actually preview the SVGs on github, so I don't really think there's a big need for it https://github.com/DHI/react-components/blob/mike-icon-migration/packages/react-components-icons/src/svg-icons/edit.svg

**Comment by rosudhi** [comment]
> I don't know how the storybook could work with icons being generated as individual react components directly to the `dist` dir.
> 
> You can actually preview the SVGs on github, so I don't really think there's a big need for it https://github.com/DHI/react-components/blob/mike-icon-migration/packages/react-components-icons/src/svg-icons/edit.svg
> 
> An example of how icons work and what props they have could be added to the main storybook instead?

this is individual svg component, and these can be used as a listing in storybook
image

as for now, i think width, height, and color is enough. and in current codes, storybook can read and show the icons, but still needs work for styling...

the looks of storybook is basically similar to this:
image
which generated based on svg files we have. and i agree, if we remove the storybook from this icon package, but again, this storybook would be ideal if we can put directly to lab component, but for now, i have no idea how to achieve that.

**Pull request closed without merging**